modules/study/src/main/PgnImport.scala

```diff
@@ -159,13 +163,14 @@ object PgnImport:
    * 7. c4 (7. c4 Nf6) (7. c4 dxc4) 7... cxd4
    * where 7. c4 appears three times
    */
-  private def removeDuplicatedChildrenFirstNode(children: Node.Children): Node.Children =
+  // TODO this could probably be refactored better or moved to scalachess
+  private def removeDuplicatedChildrenFirstNode(children: Branches): Branches =
     children.first match
       case Some(main) if children.variations.exists(_.id == main.id) =>
-        Node.Children {
+        Branches {
           main +: children.variations.flatMap { node =>
             if (node.id == main.id) node.children.nodes
-            else Vector(node)
+            else List(node)
           }
         }
       case _ => children
```

Comment:
Do you think this would be better suited in scalachess? Not sure

Reply:
I think scalachess can provide functionality for this, and maybe it should.

Also this function seems to remove children that duplicated with the main node, not if duplicated children by themself.
I mean something like: `7. d4 (7. c4 Nf6) (7. c4 dxc4) 7... cxd4`
Should it be `7. d4 (7. c4 ( 7... Nf6, 7...dxc4) 7... cxd4`. Not sure. It seems we only do the later in the study ui.

But maybe that the desired feature.